zend_abstract_interface/headers/php5/headers.c

```diff
@@ -0,0 +1,64 @@
+#include "../headers.h"
+
+#include <php.h>
+#include <zai_assert/zai_assert.h>
+
+zai_header_result zai_read_header(zai_string_view uppercase_header_name, zai_string_view *header_value TSRMLS_DC) {
+    if (!zai_string_stuffed(uppercase_header_name) || !header_value) return ZAI_HEADER_ERROR;
+
+    *header_value = ZAI_STRING_EMPTY;
+
+    zai_assert_is_upper(uppercase_header_name.ptr, "Header names must be uppercase.");
+
+    if (!PG(modules_activated) && !PG(during_request_startup)) return ZAI_HEADER_NOT_READY;
+
+    if (PG(auto_globals_jit)) {
+        // !!!
+        // This has side effects: while it does not realistically affect anybodys code - it materializes the
+        // $_SERVER array for users with auto_globals_jit On (which is observable from userland).
+        // In reality, it does not affect us much, as we anyway have that sort of side effect as part of initializing
+        // any span.
+        // As long as this function is not called with tracing disabled, this should be fine.
+        //
+        // The alternative would be calling sapi_module_struct.register_server_variables manually, but this has an
+        // unacceptable overhead as that always computes the *whole* _SERVER array, even if we just want to access
+        // a single value.
+        zend_is_auto_global(ZEND_STRL("_SERVER") TSRMLS_CC);
+    }
+
+    zval *server_var = PG(http_globals)[TRACK_VARS_SERVER];
+    if (Z_TYPE_P(server_var) != IS_ARRAY) {
```

Comment:
It looks like we added the NULL check to PHP 7+ but not here. Did you mean to add that to PHP 5?

Reply:
Yes...